gist/client.py

```diff
@@ -227,6 +228,30 @@ def elide(txt, width=terminal_width()):
     return txt
 
 
+def get_value_from_command(string):
+    """Return the value of a config option, potentially by running a command
+
+    When a config option begins with a ``!`` interpret the remaining text as a
+    shell command which when run prints the config option value to stdout.
+    Otherwise return the original string.
+
+    Argumnet:
+        string: value of an option returned from the config file.
+
+    """
+    command = string.lstrip()
+    if command[0] == '!':
+        process = subprocess.Popen(command[1:],
+                                   shell=True,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE)
+        out, err = process.communicate()
+        if process.returncode != 0:
+            raise GistError(err)
+        return out.rstrip()
```

Comment:
This looks like your assuming the `lstrip()` above has been applied to the output, but this is the result of the command, so it has not been stripped at all. If you're going to clean up the result here, use `strip()` instead.

Reply:
This `lstrip()` is only to remove a trailing new line in order not to include it in the token, there shouldn't be any unexpected white-space in the token output other than that. I could that more explicit by using `out.rstrip('\n')` instead.

The reason I actually added this was due to getting no output from what I assume was an authentication failure when a trailing `\n` is present.

Reply:
Went with `strip()` in the end.